lib/action_subscriber/middleware/error_handler.rb

```diff
@@ -8,25 +8,18 @@ def initialize(app)
       end
 
       def call(env)
-        job_mutex = ::Mutex.new
-        job_complete = ::ConditionVariable.new
+        @app.call(env)
+      rescue => error
+        logger.error "FAILED #{env.message_id}"
 
-        job_mutex.synchronize do
-          ::Thread.new do
```

Comment:
I think the reason we were wrapping these was because of the channel-killing behavior in march hare (also i think when we used to have lifeguard handle the threadpools, it would not restart crashed threads). So wrapping it in a thread avoids having an uncatchable error from killing your thread (ie java io ERRORs)